src/jdk.jdwp.agent/share/native/libjdwp/eventHelper.c

```diff
@@ -605,6 +623,35 @@ handleReportVMInitCommand(JNIEnv* env, ReportVMInitCommand *command)
     /* Why aren't we tossing this: tossGlobalRef(env, &(command->thread)); */
 }
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+static void
+handleReportVMRestoreCommand(JNIEnv* env, ReportVMRestoreCommand *command)
+{
+    PacketOutputStream out = {0};
+    jbyte suspendPolicy = command->suspendPolicy;
+
+    if (JDWP_SUSPEND_POLICY(ALL) == suspendPolicy) {
+        (void)threadControl_suspendAll();
+    } else if (JDWP_SUSPEND_POLICY(EVENT_THREAD) == suspendPolicy) {
+        (void)threadControl_suspendThread(command->thread, JNI_FALSE);
+    }
+
+    outStream_initCommand(&out, uniqueID(), 0x0,
+                          JDWP_COMMAND_SET(Event),
+                          JDWP_COMMAND(Event, Composite));
+    (void)outStream_writeByte(&out, suspendPolicy);
+    (void)outStream_writeInt(&out, 1);   /* Always one component */
+    (void)outStream_writeByte(&out, JDWP_EVENT(VM_RESTORE));
+    (void)outStream_writeInt(&out, 0);    /* Not in response to an event req. */
+
+    (void)outStream_writeObjectRef(env, &out, command->thread);
+
+    outStream_sendCommand(&out);
+    outStream_destroy(&out);
+    tossGlobalRef(env, &(command->thread));
```

Comment:
Please remove the parentheses in `&(command->thread)` (for consistency with e.g. line 685).